ee/clickhouse/sql/person.py

```diff
@@ -95,29 +102,40 @@
 PERSONS_DISTINCT_ID_TABLE_BASE_SQL = """
 CREATE TABLE {table_name} ON CLUSTER {cluster}
 (
-    id Int64,
     distinct_id VARCHAR,
     person_id UUID,
     team_id Int64,
-    is_deleted Int8 DEFAULT 0
+    _sign Int8 DEFAULT 1,
+    is_deleted Int8 ALIAS if(_sign==-1, 1, 0)
     {extra_fields}
 ) ENGINE = {engine}
 """
 
 PERSONS_DISTINCT_ID_TABLE_SQL = (
     PERSONS_DISTINCT_ID_TABLE_BASE_SQL
-    + """Order By (team_id, distinct_id, person_id, id)
+    + """Order By (team_id, distinct_id, person_id)
 {storage_policy}
 """
 ).format(
     table_name=PERSONS_DISTINCT_ID_TABLE,
     cluster=CLICKHOUSE_CLUSTER,
-    engine=table_engine(PERSONS_DISTINCT_ID_TABLE, "_timestamp", REPLACING_MERGE_TREE),
+    engine=table_engine(PERSONS_DISTINCT_ID_TABLE, "_sign", COLLAPSING_MERGE_TREE),
     extra_fields=KAFKA_COLUMNS,
     storage_policy=STORAGE_POLICY,
 )
 
-KAFKA_PERSONS_DISTINCT_ID_TABLE_SQL = PERSONS_DISTINCT_ID_TABLE_BASE_SQL.format(
+# :KLUDGE: We default is_deleted to 0 for backwards compatibility for when we drop `is_deleted` from message schema.
+#    Can't make DEFAULT if(_sign==-1, 1, 0) because Cyclic aliases error.
+KAFKA_PERSONS_DISTINCT_ID_TABLE_SQL = """
+CREATE TABLE {table_name} ON CLUSTER {cluster}
+(
+    distinct_id VARCHAR,
+    person_id UUID,
+    team_id Int64,
+    _sign Nullable(Int8),
+    is_deleted Nullable(Int8)
```

Comment:
Note on why nullable + why coalesce below: Technically the kafka table engine supports `DEFAULT` clauses. However:

1. You can't make two DEFAULTs co-dependant (e.g. `_sign DEFAULT if(is_deleted==0, 1, -1)` and `is_deleted DEFAULT if(_sign==1, 0, -1)`
2. Without nullability, the default is never called

Since this part was super finnicky, I ended up doing the "default handling"/backwards compatibility bits in the materialized view query instead.